src/fa2Interface.ml

```diff
@@ -232,21 +239,28 @@ let[@inline] fa2_run_update_operators
            { st  with
              operators =
                Ligo.Big_map.remove
-                 (op.operator, op.owner)
+                 (op.operator, op.owner, op.token_id)
                  st.operators;
            }
     )
     st
     xs
 
+let[@inline] fa2_get_balance (st, owner, token_id: fa2_state * Ligo.address * fa2_token_id): Ligo.nat =
+  let ledger = st.ledger in
+  let key = (token_id, owner) in
+  let () = assert_valid_fa2_token token_id in
```

Comment:
Note that it would have been safer to use "ensure" instead of "assert" (i.e., `ensure_valid_fa2_token`), but I saw now that our `generate-ligo.sh` script is not replacing _all_ `assert`-prefixed lines anymore so it should be OK either way :slightly_smiling_face:

Reply:
Since you have this assertion as part of `fa2_get_balance`, perhaps you can now remove the same one in `fa2_run_balance_of` (line 262 below)?

Reply:
I agree with both comments. 

As about the first comment, I think we are overdue for a small rethink of `failwith`, `Ligo.failwith` and assertions. Probably soon, because TZIP-16 spec also has a field called `errors` that requires us to have structured error code to message mapping.

Reply:
Right, good point. Our error codes though are still a little fluid so that would easily get out of sync. I certainly agree though that we gotta revisit the `failwith`/`assert` situation.